images/img-builder.sh

```diff
@@ -0,0 +1,79 @@
+#!/bin/bash 
+
+# This is PoC for building images without requiring admin capabilities (CAP_SYS_ADMIN)
+
+rm -rf ./*.part disk.raw grub_efi.cfg squashfs root .luet.yaml
+
+set -e
+
+truncate -s $((20*1024*1024)) efi.part
+
+# Create the EFI partition FAT16 and include the EFI image and a basic grub.cfg
+mkfs.fat -F16 -n EFI efi.part
+mmd -i efi.part ::EFI
+mmd -i efi.part ::EFI/BOOT
+mcopy -i efi.part /usr/share/grub2/x86_64-efi/grub.efi ::EFI/BOOT/bootx64.efi
+
+# Creat the EFI grub.cfg pointing the configs in COS_RECOVERY volume
+cat << 'EOF' > grub_efi.cfg
+search.fs_label COS_RECOVERY root
+set root=($root)
+set prefix=($root)/grub2
+configfile ($root)/grub2/grub.cfg
+EOF
+
+# Copy grub.cfg in EFI partition
+mcopy -i efi.part grub_efi.cfg ::EFI/BOOT/grub.cfg
+
+# Create a luet config for local repositories
+cat << EOF > .luet.yaml
+repositories:
+  - name: cOS
+    enable: true
+    urls:
+      - build
+    type: disk
+EOF
+
+# Create root-tree for COS_RECOVERY
+mkdir -p squashfs
+# `luet install` is the only step requiring root
+luet install --system-target squashfs -y system/cos
```

Comment:
mmh, for consistency, would be nice to read those directly from the `manifest.yaml` file, so we can actually specify all of them in one place